nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -337,6 +343,43 @@ impl<'a> BlueprintBuilder<'a> {
                 || commissioned_sled_ids.contains(sled_id)
         });
 
+        // If we have the clickhouse cluster setup enabled via policy and we
+        // don't yet have a `ClickhouseClusterConfiguration`then we must create
+        // one and feed it to our `ClickhouseAllocator`
+        let clickhouse_allocator = if input.clickhouse_cluster_enabled() {
+            let parent_config = parent_blueprint
+                .clickhouse_cluster_config
+                .clone()
+                .unwrap_or_else(|| {
+                    info!(
+                        log,
+                        concat!(
+                            "Clickhouse cluster enabled by policy: ",
+                            "generating initial 'ClickhouseClusterConfig' ",
+                            "and 'ClickhouseAllocator'"
+                        )
+                    );
+                    ClickhouseClusterConfig::new(OXIMETER_CLUSTER.to_string())
+                });
+            Some(ClickhouseAllocator::new(
+                log.clone(),
+                parent_config,
+                inventory.latest_clickhouse_keeper_membership(),
+            ))
+        } else {
+            if parent_blueprint.clickhouse_cluster_config.is_some() {
```

Comment:
If this gets discarded we will have to expunge *all* the existing `ClickhouseKeeper` and `ClickhouseServer` zones. 

This is not really an issue for this PR since there is no current way to actually set the clickhouse policy yet.

Reply:
I should open an issue for this and probably add a comment regarding it to the code.

Reply:
Also possible, that I should implement it in this PR with a test. Shouldn't be hard.

Reply:
I'm working on implementing this in the current PR right now.

Reply:
Implemented in [ae380df](https://github.com/oxidecomputer/omicron/pull/6627/commits/ae380dff525a72028dce865bba57e210f47d70f5)

I will add a test here shortly.

Reply:
Test added in [592608d](https://github.com/oxidecomputer/omicron/pull/6627/commits/592608d2fd1468e1a9cd16153987f0d244d3c7dc)